src/instructlab/lab.py

```diff
@@ -876,6 +876,10 @@ def convert(self, value, param, ctx) -> "torch.device":
         # pylint: disable=C0415
         # Function local import, import torch can take more than a second
         # Third Party
+        if value.startswith("cuda:"):
+            index = value.split(":")[1]
+            os.environ["CUDA_VISIBLE_DEVICES"]=str(index)
```

Comment:
Why is this env var needed?

Reply:
in linux_train(): model = model.to(device) does not work to use specific gpu.

set environment CUDA_VISIBLE_DEVICES=<index> can make pytorch only see the <index> gpu.

refer https://github.com/huggingface/transformers/issues/25321

Reply:
https://huggingface.co/docs/transformers/main/en/perf_train_gpu_many#order-of-gpus